app/controllers/general.php

```diff
@@ -40,9 +43,136 @@
 Config::setParam('cookieDomain', 'localhost');
 Config::setParam('cookieSamesite', Response::COOKIE_SAMESITE_NONE);
 
+function router(App $utopia, Database $dbForConsole, SwooleRequest $swooleRequest, Request $request, Response $response)
+{
+    $utopia->getRoute()->label('error', __DIR__ . '/../views/general/error.phtml');
+
+    $host = $request->getHostname() ?? '';
+
+    $route = Authorization::skip(
+        fn () => $dbForConsole->find('rules', [
+            Query::equal('domain', [$host]),
+            Query::limit(1)
+        ])
+    )[0] ?? null;
+
+    if ($route === null) {
+        $mainDomain = App::getEnv('_APP_DOMAIN', '');
+
+        if ($mainDomain === 'localhost') {
+            throw new AppwriteException(AppwriteException::GENERAL_SERVER_ERROR, 'Please configure domain environment variables before using Appwrite outside of localhost.');
```

Comment:
This should probably be an error of its own type. Maybe `ROUTER_DOMAIN_NOT_CONFIGURED` ?